workflow/controller/operator.go

```diff
@@ -1338,6 +1338,16 @@ func (woc *wfOperationCtx) executeTemplate(nodeName string, orgTmpl wfv1.Templat
 		if err != nil {
 			return woc.markNodeError(retryNodeName, err), err
 		}
+		if lastChildNode != nil && lastChildNode.Pending() && processedTmpl.GetType() == wfv1.TemplateTypeContainer {
```

Comment:
Why do we only look at `wfv1.TemplateTypeContainer`? Both `wfv1.TemplateTypeScript` and `wfv1.TemplateTypeResource` also execute Pods.

Reply:
@simster7, is it possible to specify limits for these types? It's actually a requirement, as if you have limits on a namespace, it's impossible to submit a pod without limits defined. There is a workaround, thouth, LimitRange object to set default values. Also, I'd suggest if this functionality is required (to retry these types on submission failure), to make it another story.

Reply:
> is it possible to specify limits for these types?

Do you mean resource limits on the k8s? Both of those template types execute pods (to k8s there is no difference between a `TemplateTypeContainer` and a `TemplateTypeScript`).

> Also, I'd suggest if this functionality is required (to retry these types on submission failure), to make it another story.

Why do you think so? From my perspective retrying Script templates is the same as retrying Container templates.